[CI:DOCS] Add documentation on short-names (#10249)
containers/podman: docs/source/markdown/podman-pull.1.md

## DESCRIPTION

> Copies an image from a registry onto the local machine. **podman pull** pulls an

Reviewer: ```suggestion
Copies an image from a registry onto the local machine. The **podman pull** command pulls an
```

> image from the default container registry is not specified in the command line argument, these are called `short-names`. Podman will prompt the user for the specific container registry to pull the image from.  If an image tag is not specified, **podman pull** defaults to the image with the **latest** tag (if it exists) and pulls it. After the image is pulled, podman will print the full image ID.  **podman pull** can also pull an image using its digest **podman pull** *image*@*digest*. **podman pull** can be used to pull images from archives and local storage using different transports.

Reviewer: ```suggestion
image.  If the image reference in the command line argument does not contain a registry, it is referred to as a`short-name` reference. If the image is a 'short-name', Podman will prompt the user for the specific container registry to pull the image from, if an alias for the short-name has not been specified in the short-name-aliases.conf.  If an image tag is not specified, **podman pull** defaults to the image with the **latest** tag (if it exists) and pulls it. After the image is pulled, podman will print the full image ID.  **podman pull** can also pull an image using its digest **podman pull** *image*@*digest*. **podman pull** can be used to pull images from archives and local storage using different transports.
```

## FILES

**short-name-aliases.conf** (`/var/cache/containers/short-name-aliases.conf`, `$HOME/.cache/containers/short-name-aliases.conf`)

When users specify images that do not include the container registry where the
image is stored, this is called a short name. The use of unqualified-search registries entails an ambiguity as it is unclear from which registry a given image, referenced by a short name, may be pulled from.

> Using short names is subject to the risk of hitting squatted registry namespaces. If the unqualified-search registries are set to ["registry1.com",  "registry2.com"] an attacker may take over a namespace of registry1.com such that an image may be pulled from registry1.com instead of the in‐tended source registry2.com.

Reviewer 2: in-tended -> intended

Author: Fixed

> While it is highly recommended to always use fully-qualified image references, existing deployments using short names may not be easily changed. To circumvent the aforementioned ambiguity, so called short-name aliases can be configured that point to a fully-qualified image reference. Distributions often ship a default shortnames.conf expansion file in /etc/containers/registries.conf.d/ directory. Adminsitrarotes can use this directory to add their own local short-name expansion files.

Reviewer 3: Typo, should be `Administrators`

Author: Oops.

When pulling an image, if the user does not specify the complete registry, container engines attempt to expand the short-name into a full name. If the command is executed with a tty, the user will be prompted to select a registry from the

> default list unqualified registies defined in registries.conf. The users selection is then stored in a cache file to be used in all future short-name expansions. Rootfull short-names are stored in /var/cache/containers/short-name-aliases.conf. Rootless short-names are stored in the $HOME/.cache/containers/short-name-aliases.conf file.

Reviewer 3: The spelling is `rootful`, one el. (A few double-el rootfulls have snuck in since the last time we cleaned up, though)

Reviewer 2: registies -> registries

The users selection -> The user's selection

Author: fixed

> For more information on short-names, see `containers-registries.conf(5)

Reviewer 4: ```suggestion
For more information on short-names, see `containers-registries.conf(5)`
```
